### ISSUE TITLE
Contour tool not circling around mesh

### ISSUE THREAD

**Issue by kobold21** [CGCookie/retopoflow#495]
I'm newer to blender and 3D modeling, and I've started retopology work on a high poly. I've installed the tool and started work, but the contours tool will not cover the perimeter of a bit of mesh. The other tools work fine, but this one is giving me trouble. The tool seems to simply link the top and bottom of the created mesh by going through my model, which is weird. It works on lower poly meshes, but not this one that I've sculpted for a while.
![contours problem](https://user-images.githubusercontent.com/40246612/41371424-428b3554-6f18-11e8-86e4-376c30ac62db.png)

**Comment by vxlcoder** [comment]
Thanks for the report, @kobold21.

You are using Contours from RetopoFlow 1.3, and we have rewritten the tools from the ground up to be more robust, consistent, etc.  The new version of RF is still in beta at the moment (2.0 beta), but when we officially release it, the RF 1.3 tools will be removed due to them being more difficult to debug and update.

Can you try using the new RF 2.0 Contours to see if it is giving you the same problems?  If you have purchased RF through the Blender Market, you should have received an updated version a couple of months ago.  If you're git savvy, clone/checkout the `release/2.0.0` branch.

Note: We have made changes in 2.0 that significantly differ in how you use RF.  Be sure to peruse the Quick Start Guide.

**Comment by kobold21** [comment]
Thanks for the quick reply!
I installed the update through blender and started drawing contours, and while the edges do not connect through the mesh, the created contours still do not go around the whole mesh. 
![contours problem 2 0](https://user-images.githubusercontent.com/40246612/41373976-e5fbbac2-6f1f-11e8-9fb3-aac0178e7a37.png)

**Comment by vxlcoder** [comment]
Interesting.  Could you by chance post your .blend file?

**Comment by kobold21** [comment]
Here you go! Pardon the quality; I'm still learning. The leftmost render layers are the models, the bottom are the high-poly. The low body poly in progress is by the traditional method. I should also mention that I had retopoflow suggested I scale down the model, but this specific version of the file is at full size.
[SpoonLowPolySplit.zip](https://github.com/CGCookie/retopoflow/files/2099643/SpoonLowPolySplit.zip)

**Comment by vxlcoder** [comment]
Thanks, @kobold21!

The scaling is just to help with visualizing what is above/below the source, but it has no effect on functionality.

A few things I've noticed with your model/blend file.  Below is a short list.

- The center of target mesh is not the same as your source, which means that mirroring will not work correctly.  If you apply the Translation Transformation, it will move the target center to the origin, which is where the source center is located, so they mirroring will work (without needing the shrinkwrap modifier).
- You have a keyframe set for transformation on the target.  RF should work just fine as it is, but I haven't tested in thoroughly enough.  You can safely delete the keyframe in the graph editor.
- You have Lock to Cursor enabled.  Do you normally work that way?  I haven't done much with that Blender feature, so I haven't tested it much.  Does navigating in RF work OK with that enabled?

Now, for the solution to your problem: Your source mesh seems to have a lot of non-manifold edges.  In other words, you have a lot (24k!) of duplicate vertices/edges.  Removing these seems to have fixed your problem. (edit source mesh, select all verts, remove doubles)

Please let me know if you continue to have problems!

**Comment by kobold21** [comment]
As for your questions:

- Thanks! I forgot about moving the origin. I read somewhere that shrinkwrap would help retopology if your new mesh has a subsurface simple subdivide.
- The keyframe came from a stray click on my end!
- I've been using the cursor lock so I can pivot around small details and to control to where I zoom in. It works perfectly fine with RF.

As for your solution, you were completely correct. I had applied a few modifiers to smooth and started sculpting before I learned I needed to lower the polygon count if I intended to use the model as a game asset. After removing the duplicate vertices, RF worked perfectly! Thank you so very much for your help!
